server/src/main/java/org/elasticsearch/common/xcontent/XContentFieldFilter.java

```diff
@@ -73,8 +73,8 @@ static XContentFieldFilter newFieldFilter(String[] includes, String[] excludes)
             };
         } else {
             final XContentParserConfiguration parserConfig = XContentParserConfiguration.EMPTY.withFiltering(
-                Set.of(includes),
-                Set.of(excludes),
+                Set.copyOf(Arrays.asList(includes)),
```

Comment:
Given that we can have duplicated entries, although whether they are allowed or not depends on the implementation, I wonder if we should change the signature of withFiltering here, and take a Collection instead.